app/main.py

```diff
@@ -1,2 +1,62 @@
+from typing import Any
+
+
 class Dictionary:
-    pass
+    class Node:
+        def __init__(self, key: Any, value: Any) -> None:
+            self.key = key
+            self.hash = hash(key)
+            self.value = value
+
+        def __repr__(self) -> str:
+            return f"({self.key}, {self.hash}, {self.value})"
+
+    def __init__(self,
+                 initial_capacity: int = 8,
+                 load_factor: float = 2 / 3) -> None:
+        self.capacity = initial_capacity
+        self.load_factor = load_factor
+        self.table = [None] * initial_capacity
+
+    def index(self, key: Any) -> int:
+        return hash(key) % self.capacity
+
+    def resize(self) -> None:
+        old_table = self.table
+        self.capacity = self.capacity * 2
+        self.table = [None] * self.capacity
+
+        for node in old_table:
+            if node is not None:
+                self.__setitem__(node.key, node.value)
+
+    def calculate_size(self) -> int:
+        return sum(1 for node in self.table if node is not None)
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        if (self.calculate_size() + 1) / self.capacity >= self.load_factor:
```

Comment:
The condition for checking if resizing is needed is incorrect. It should be `self.calculate_size() / self.capacity >= self.load_factor` instead of `(self.calculate_size() + 1) / self.capacity >= self.load_factor`. This ensures that the dictionary resizes at the correct time.